pkg/storkctl/action.go

```diff
@@ -20,99 +20,144 @@ const (
-				printMsg(fmt.Sprintf("Started failover for namespace %v", namespace), ioStreams.Out)
-				printMsg(getDescribeActionMessage(&action), ioStreams.Out)
+			} else {
+				// Branch 3: Failover all the namespaces being migrated by the given migrationSchedule
+				namespaceList = migrationNamespaces
+			}
+			actionType := storkv1.ActionTypeFailover
+			actionParameters := storkv1.ActionParameter{
+				FailoverParameter: &storkv1.FailoverParameter{
+					FailoverNamespaces:         namespaceList,
+					MigrationScheduleReference: referenceMigrationSchedule,
+					DeactivateSource:           !skipDeactivateSource,
+				},
+			}
+			actionName := getActionName(actionType, referenceMigrationSchedule)
+			action, err := createActionCR(actionName, migrationScheduleNs, actionType, actionParameters)
+			migrationScheduleName := migrationScheduleNs + "/" + referenceMigrationSchedule
+			if err != nil {
+				util.CheckErr(fmt.Errorf("failed to start failover for migrationSchedule %v : %v", migrationScheduleName, err))
+				return
 			}
+			printMsg(fmt.Sprintf("Started failover for migrationSchedule %v", migrationScheduleName), ioStreams.Out)
+			printMsg(getActionStatusMessage(action), ioStreams.Out)
 		},
 	}
-	return failoverCommand
+	performFailoverCommand.Flags().BoolVar(&skipDeactivateSource, "skip-deactivate-source", false, "If present, applications in the source cluster will not be scaled down as part of the failover.")
+	performFailoverCommand.Flags().StringVarP(&referenceMigrationSchedule, "migration-reference", "m", "", "Specify the migration schedule to failover. Also specify the namespace of this migrationSchedule using the -n flag")
+	performFailoverCommand.Flags().StringSliceVar(&includeNamespaceList, "include-namespaces", nil, "Specify the comma-separated list of subset namespaces to be failed over. By default, all namespaces part of the migrationSchedule are failed over")
+	performFailoverCommand.Flags().StringSliceVar(&excludeNamespaceList, "exclude-namespaces", nil, "Specify the comma-separated list of subset namespaces to be skipped during the failover. By default, all namespaces part of the migrationSchedule are failed over")
+	return performFailoverCommand
 }
 
-func isActionIncomplete(action *storkv1.Action) bool {
-	return action.Status.Status == storkv1.ActionStatusScheduled || action.Status.Status == storkv1.ActionStatusInProgress
-}
-
-// check if there is already an Action scheduled or in-progress
-func getAnyIncompleteAction(namespace string) *storkv1.Action {
-	actionList, err := storkops.Instance().ListActions(namespace)
-	if err != nil {
-		util.CheckErr(err)
-		return nil
+// given the name, namespace, actionType and actionParameters create an Action CR
+func createActionCR(actionName string, namespace string, actionType storkv1.ActionType, actionParameters storkv1.ActionParameter) (*storkv1.Action, error) {
+	action := storkv1.Action{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      actionName,
+			Namespace: namespace,
```

Comment:
Should we put an annotation "created-by: stork" to figure out if it has been created by storkctl ?

Reply:
Since we are not adding such an annotation in other migration objects clusterpair, migrationschedule 
I think we can skip this for now.
If we decide in the future we can make the change across all objects.